includes/class-micropub-endpoint.php

```diff
@@ -0,0 +1,1201 @@
+			}
+			foreach ( static::mp_to_wp( array( 'properties' => $replace ) )
+					as $name => $val ) {
+				$args[ $name ] = $val;
+			}
+		}
+
+		// delete
+		$delete = static::$input['delete'];
+		if ( $delete ) {
+			if ( is_assoc_array( $delete ) ) {
+				if ( array_diff( array_keys( $delete ), array( 'category', 'syndication' ) ) ) {
+					static::error( 400, 'can only delete individual values from category and syndication; other properties not supported' );
+				}
+				$delete_args = static::mp_to_wp( array( 'properties' => $delete ) );
+				if ( $delete_args['tags_input'] ) {
+					$args['tags_input'] = array_diff(
+						$args['tags_input'] ?: array(),
+						$delete_args['tags_input']
+					);
+				}
+				if ( $delete_args['post_category'] ) {
+					$args['post_category'] = array_diff(
+						$args['post_category'] ?: array(),
+						$delete_args['post_category']
+					);
+				}
+			} elseif ( is_array( $delete ) ) {
+				foreach ( static::mp_to_wp( array( 'properties' => array_flip( $delete ) ) )
+					as $name => $_ ) {
+					$args[ $name ] = null;
+				}
+				if ( in_array( 'category', $delete, true ) ) {
+					wp_set_post_tags( $post_id, '', false );
+					wp_set_post_categories( $post_id, '' );
+				}
+			} else {
+				static::error( 400, 'delete must be an array or object' );
+			}
+		}
+
+		// tell WordPress to preserve published date explicitly, otherwise
+		// wp_update_post sets it to the current time
+		$args['edit_date'] = true;
+
+		// Generate Post Content
+		$post_content = self::get( $args, 'post_content', '' );
+		$post_content = apply_filters( 'micropub_post_content', $post_content, static::$input );
+		if ( $post_content ) {
+			$args['post_content'] = $post_content;
+		}
+
+		// Store metadata from Microformats Properties
+		$args = static::store_mf2( $args );
+		$args = static::store_geodata( $args );
+
+		if ( WP_DEBUG ) {
+			error_log( 'wp_update_post with args: ' . wp_json_encode( $args ) );
+		}
+
+		kses_remove_filters();
+		static::check_error( wp_update_post( $args, true ) );
+		kses_init_filters();
+
+		static::default_file_handler( $post_id );
+		return $args;
+	}
+
+	private static function handle_authorize_error( $code, $msg ) {
+		$home = untrailingslashit( home_url() );
+		if ( 'http://localhost' === $home ) {
+			error_log(
+				'WARNING: ' . $code . ' ' . $msg .
+				". Allowing only because this is localhost.\n"
+			);
+			return;
+		}
+		static::respond( $code, $msg );
+	}
+
+	private static function default_post_status() {
+		$option = get_option( 'micropub_default_post_status', '' );
+		return empty( $option ) ? ( MICROPUB_DRAFT_MODE ? 'draft' : 'publish' ) : $option;
+	}
+
+	private static function post_status( $mf2 ) {
+		$props = $mf2['properties'];
+		// If both are not set immediately return
+		if ( ! isset( $props['post-status'] ) && ! isset( $props['visibility'] ) ) {
+			$status = self::default_post_status();
+			if ( ! in_array( $status, array( 'publish', 'draft', 'private' ), true ) ) {
```

Comment:
looks like this changed. old code is below. change looks fine, except this `if` (lines 538-540) is entirely a noop, since both paths `return $status`. mind deleting it?

```php
private static function default_post_status() {
	return MICROPUB_DRAFT_MODE ? 'draft' : 'publish';
}

private static function post_status( $mf2 ) {
	$props = $mf2['properties'];
	// If both are not set immediately return
	if ( ! isset( $props['post-status'] ) && ! isset( $props['visibility'] ) ) {
		$status = get_option( 'micropub_default_post_status', static::default_post_status() );
		if ( ! in_array( $status, array( 'publish', 'draft', 'private' ), true ) ) {
			return static::default_post_status();
		}
		return $status;
	}
	...
```

Reply:
I made some adjustments to this to move more of the logic into the default function.